redfish-core/lib/environment_metrics.hpp

```diff
@@ -0,0 +1,372 @@
+
+inline void
+    getPowerLimitWatts(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code ec,
+                    const std::vector<
+                        std::pair<std::string, std::variant<uint32_t, bool>>>&
+                        properties) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "Can't get Power Limit Watts!";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            bool enabled = false;
+            uint32_t powerCap = 0;
+            for (const std::pair<std::string, std::variant<uint32_t, bool>>&
+                     property : properties)
+            {
+                if (!property.first.compare("PowerCap"))
+                {
+                    const uint32_t* u = std::get_if<uint32_t>(&property.second);
+                    if (u)
+                    {
+                        powerCap = *u;
+                    }
+                }
+                else if (!property.first.compare("PowerCapEnable"))
+                {
+                    const bool* b = std::get_if<bool>(&property.second);
+
+                    if (b)
+                    {
+                        enabled = *b;
+                    }
+                }
+            }
+            nlohmann::json& value =
+                asyncResp->res.jsonValue["PowerLimitWatts"]["SetPoint"];
+            if (enabled)
+            {
+                value = powerCap;
+            }
+        },
+        "xyz.openbmc_project.Settings",
+        "/xyz/openbmc_project/control/host0/power_cap",
+        "org.freedesktop.DBus.Properties", "GetAll",
+        "xyz.openbmc_project.Control.Power.Cap");
+}
+
+inline void
+    setPowerLimitWatts(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       uint32_t powerCap)
+{
+    BMCWEB_LOG_DEBUG << "Set Power Limit Watts";
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, powerCap](const boost::system::error_code ec,
+                              const std::variant<bool>& powerCapEnable) {
+            if (ec)
+            {
+                messages::internalError(asyncResp->res);
+                BMCWEB_LOG_ERROR << "powerCapEnable Get handler: Dbus error "
+                                 << ec;
+                return;
+            }
+            const bool* b = std::get_if<bool>(&powerCapEnable);
+            if (b == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                BMCWEB_LOG_ERROR << "Fail to get PowerCapEnable status ";
+                return;
+            }
+            if (!(*b))
+            {
+                messages::actionNotSupported(
+                    asyncResp->res,
+                    "Setting LimitInWatts when PowerLimit feature is disabled");
+                BMCWEB_LOG_ERROR << "PowerLimit feature is disabled ";
```

Comment:
How do I enable this? 
See comments upstream, I think we need to use  "ControlMode": "Enabled"

Reply:
Done